src/main/scala/org/marvin/executor/EngineExecutorApp.scala

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright [2017] [B2W Digital]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.marvin.executor
+
+import java.io.FileNotFoundException
+
+import akka.actor.{ActorRef, ActorSystem, Props}
+import com.fasterxml.jackson.core.JsonParseException
+import com.github.fge.jsonschema.core.exceptions.ProcessingException
+import com.typesafe.config.{Config, ConfigFactory}
+import grizzled.slf4j.Logger
+import org.marvin.exception.MarvinEExecutorException
+import org.marvin.executor.actions.{BatchAction, OnlineAction, PipelineAction}
+import org.marvin.executor.api.{GenericAPI, GenericAPIFunctions}
+import org.marvin.executor.manager.ExecutorManager
+import org.marvin.executor.statemachine.{PredictorFSM, Reload}
+import org.marvin.model.EngineMetadata
+import org.marvin.util.{ConfigurationContext, JsonUtil}
+
+import scala.io.Source
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+
+object EngineExecutorApp {
+  def main(args: Array[String]): Unit = {
+
+    val app = new EngineExecutorApp()
+
+    // setup actors system before start server
+    app.setupGenericAPI()
+
+    // if requested enable remote administration
+    app.setupAdministration()
+
+    // start generic api server
+    app.start()
+  }
+}
+
+class EngineExecutorApp {
+
+  var vmParams: Map[String, Any] = getVMParameters
+  var executorManager: ActorRef = _
+  var api: GenericAPIFunctions = _
+
+  lazy val log = Logger[this.type]
+
+  def setupConfig(): Config = {
+
+    log.info("Configuring engine executor app...")
+
+    if (vmParams("enableAdmin").asInstanceOf[Boolean]) {
+      val configuration = """
```

Comment:
Will the application.conf be ignored in this case?

Reply:
No...In this case a merge will be done.

* ConfigFactory.load() -> Loads default configuration from "application.conf" file